File: legacy/src/main/java/com/redbox/domain/article/dto/ArticleResponse.java
package com.redbox.domain.article.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.redbox.domain.article.entity.Article;
import lombok.Getter;

import java.time.LocalDate;

@Getter
@JsonDeserialize
public class ArticleResponse {

    private Long articleNo;
    private String subject;
    private String url;
    private String source;
    private LocalDate createdDate;

    @JsonCreator
    public ArticleResponse() {
    }

    public ArticleResponse(Article article) {
        this.articleNo = article.getId();
        this.subject = article.getSubject();
        this.url = article.getArticleUrl();
        this.source = article.getSource();
        this.createdDate = article.getCreatedAt().toLocalDate();
    }
}

File: migration/src/main/kotlin/com/redbox/domain/community/article/dto/ArticleResponse.kt
package com.redbox.domain.community.article.dto

import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.redbox.domain.community.article.entity.Article
import java.time.LocalDate

@JsonDeserialize
@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
data class ArticleResponse(
    val articleNo: Long,
    val subject: String,
    val url: String,
    val source: String,
    val createdDate: LocalDate
) {
    companion object {
        fun fromArticle(article: Article): ArticleResponse {
            return ArticleResponse(
                articleNo = article.id ?: throw IllegalStateException("Notice not found"),
                subject = article.subject ?: "No Title",
                url = article.articleUrl ?: "No Content",
                source = article.source ?: "No Source",
                createdDate = article.createdAt?.toLocalDate() ?: LocalDate.now(),
            )
        }
    }
}

